File: Client/src/services/TaskService.js
import axios from "axios"
import { BACKEND_URL } from "../config"

const baseUrl = `${BACKEND_URL}/tasks`

async function getAll() {
  const response = await axios.get(baseUrl)
  return response.data
}

async function post(title) {
  const response = await axios.post(baseUrl, { title, completed: false })
  return response.data
}

async function update(id, title) {
  const response = await axios.patch(`${baseUrl}/${id}`, { title })
  return response.data
}

async function remove(id) {
  await axios.delete(`${baseUrl}/${id}`)
}

const taskService = { post, update, getAll, remove }
export default taskService

File: api/Services/TaskService.cs
using Microsoft.EntityFrameworkCore;
using OnlyTodo.Data;
using OnlyTodo.Models;

namespace OnlyTodo.Services;

public class TaskService
{
    private readonly OnlyTodoContext _context;

    public TaskService(OnlyTodoContext context)
    {
        _context = context;
    }

    public Task<List<TodoTask>> GetAllAsync()
    {
        return _context.Tasks.ToListAsync();
    }

    public async Task<TodoTask?> FindAsync(Guid id)
    {
        return await _context.Tasks.FindAsync(id);
    }

    public async Task<TodoTask> AddAsync(TodoTask task)
    {
        // Don't trust the given guid
        task = new TodoTask(Guid.NewGuid(), task.Title, task.Completed);

        _context.Add(task);
        await _context.SaveChangesAsync();

        return task;
    }

    public async Task<TodoTask> UpdateAsync(TodoTask taskUpdates)
    {
        TodoTask task = _context.Tasks.Find(taskUpdates.Id) ?? throw new Exception();

        _context.Update(task);
        await _context.SaveChangesAsync();

        return task;
    }

    public async Task RemoveAsync(Guid id)
    {
        TodoTask task = _context.Tasks.Find(id) ?? throw new NullReferenceException();

        _context.Remove(task);
        await _context.SaveChangesAsync();
    }
}
